src/commands/run.cr

```diff
@@ -0,0 +1,41 @@
+require "./command"
+
+module Shards
+  module Commands
+    class Run < Command
+      def run(targets, options, run_options)
+        if spec.targets.empty?
+          raise Error.new("Targets not defined in #{SPEC_FILENAME}")
+        end
+
+        # when more than one target was specified
+        if targets.size > 1
+          raise Error.new("Error please specify only one target. If you meant to pass arguments you may use 'shards run target -- args'")
+        end
+
+        # when no target was specified
+        if targets.empty?
+          if spec.targets.size > 1
+            raise Error.new("Error please specify the target with 'shards run target'")
+          else
+            name = spec.targets.first.name
+          end
+        else
+          name = targets.first
+        end
+
+        if target = spec.targets.find { |t| t.name == name }
+          Commands::Build.run(path, [target.name], options)
+
+          Log.info { "Executing: #{target.name} #{run_options.join(' ')}" }
+          status = Process.run(File.join(Shards.bin_path, target.name), args: run_options, output: Process::Redirect::Inherit, error: Process::Redirect::Inherit)
```

Comment:
```suggestion
          status = Process.run(File.join(Shards.bin_path, target.name), args: run_options, output: :inherit, error: :inherit)
```

Reply:
Thought about this, however decided to follow the same pattern/style used in build command:

https://github.com/crystal-lang/shards/blob/master/src/commands/build.cr#L47

Could change if really needed.

Reply:
Either way is good